core/src/test/java/kafka/test/MockController.java

```diff
@@ -410,10 +410,16 @@ public CompletableFuture<UpdateFeaturesResponseData> updateFeatures(
         throw new UnsupportedOperationException();
     }
 
+    boolean lastCreatePartitionsValidateOnly = false;
```

Comment:
This is a little strange. Would it be enough to check whether the partition creation happened or not?

Reply:
Added a new unit test and remove `lastCreatePartitionsValidateOnly`